astropy/utils/data_info.py

```diff
@@ -1,6 +1,13 @@
 # -*- coding: utf-8 -*-
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
+"""
+This module contains functions and methods that relate to the DataInfo
+class which provides a container class for informational attributes
+as well as summary info methods.  This is currently used in the Table
```

Comment:
I think this does it a bit of disservice!  It is already useful in other context -- e.g., see https://github.com/astropy/astropy/pull/4042#issuecomment-144082795
How about
```
This is attached to all container classes in astropy
(Quantity, SkyCoord, Time) and is explicitly used in the Table
class, where it allows those containers to carry a name, etc.
```